src/Avalonia.Xaml.Interactions.Custom/Core/DisposableAction.cs

```diff
@@ -0,0 +1,27 @@
+
+namespace Avalonia.Xaml.Interactions.Custom;
+/// <summary>
+/// Represents an <see cref="System.IDisposable"/> that executes a specified action when disposed.
+/// </summary>
+/// <param name="dispose">The action to execute when disposed.</param>
+public class DisposableAction(System.Action dispose) : System.IDisposable
```

Comment:
It can be internal, unless you want to keep it public for custom third party behaviors/triggers.